test/classes.coffee

```diff
@@ -1808,3 +1808,19 @@ test 'Bound method of immediately instantiated class with expression base class
 
   {derivedBound} = a
   eq derivedBound(), 3
+
+test "#4591: super.x.y, super['x'].y", ->
```

Comment:
@GeoffreyBooth added tests for the cases you outlined

@connec that's a more realistic use case than anything I thought of, I just noticed the hole in the grammar